ext/zip/php_zip.c

```diff
@@ -2662,6 +2663,10 @@ static ZIPARCHIVE_METHOD(extractTo)
 									RETURN_FALSE;
 								}
 								break;
+							case IS_REFERENCE:
+								zval_file = Z_REFVAL_P(zval_file);
+								goto try_again;
+							EMPTY_SWITCH_DEFAULT_CASE();
```

Comment:
As far as I can tell this is a user-provided array. What ensures that that it can only contain int/string?

Reply:
Oh, right! Silently ignoring unsupported types might not have been the best idea, though.